### ISSUE TITLE
Bug: user.projects is sometimes an object

### ISSUE THREAD

**Issue by myieye** [sillsdev/languageforge-lexbox#88]
When I log in as "manager" I can navigate to the "Sena 3" project just fine. The natural login process correctly sets the state of user.projects to an array with 1 project.

If I then refresh the page it blows up, because user.projects is a project object.

A unit test documents that the "proj" claim in our jwt token is an object. I'm guessing this issue is just some unfinished business:

![image](https://github.com/sillsdev/languageforge-lexbox/assets/12587509/caa649d9-278d-47ba-b8af-27d3438dafa6)

**Comment by hahn-kev** [comment]
nice catch, so the reason is because of how jwt gets turned into an array of claims and back again.

for a jwt like this:
```json
{
"sub": "kevin",
"project": [
    {id:1},
    {id:2}
  ]
}
```
you will get 3 claims:
- key: sub, value: kevin
- key: project, value: {id:1}
- key: project, value: {id:2}

so that's why the code in the test is valid, because it's verifying that it's working as expected. If there were 2 projects in the test then there would be 2 claims with the same name.

As for the error I'm not sure. Somewhere a jwt is getting out with some bad values. I suspect the issue is [here](https://github.com/sillsdev/languageforge-lexbox/blob/e1d22b85371e085d1b9ef9cafe69cb0ee47b7fbd/backend/LexBoxApi/Auth/JwtTicketDataFormat.cs#L47) in the JwtTicketDataFormat where the Principal is getting written to a jwt cookie. We probably need to modify the claims as they get set on the subject for the `SecurityTokenDescriptor` that gets used to create the jwt.

**Comment by hahn-kev** [comment]
Should be fixed in 5aec6100
